cura/Settings/MachineManager.py

```diff
@@ -894,8 +894,8 @@ def forceUpdateAllSettings(self) -> None:
 
     @pyqtSlot(int, bool)
     def setExtruderEnabled(self, position: int, enabled: bool) -> None:
-        if self._global_container_stack is None:
-            Logger.log("w", "Could not find extruder on position %s", position)
+        if self._global_container_stack is None or position not in self._global_container_stack.extruders:
```

Comment:
I don't think this will work. Position is an integer and self._global_container_stack.extruders is a dict with a string as the key